File: Median of BST[29-July-2023]/Naive Approach/solution.cpp
void inorder(Node* root, vector<int>& list) {
    if(root==nullptr) return;
    inorder(root->left,list);
    list.push_back(root->data);
    inorder(root->right,list);
}
float findMedian(Node* root) {
    vector<int> list;
    inorder(root,list);
    int n=list.size();
    if(n%2==0) {
        int sum=list[n/2-1]+list[n/2];
        return sum%2==0?sum/2:(float)((double)sum/2);
    } else {
        return list[(n+1)/2-1];
    }
}

File: Median of BST[29-July-2023]/Naive Approach/solution.py
def findMedian(root):
    def inorder(root,elements):
        if(root==None): return root
        
        inorder(root.left,elements)
        elements.append(root.data)
        inorder(root.right,elements)
        
    elements=[]
    
    inorder(root,elements)
    
    n=len(elements)
    
    if(n%2==0): return (elements[n//2]+elements[n//2-1])/2 if (elements[n//2]+elements[n//2-1])%2==1 else int((elements[n//2]+elements[n//2-1])/2)

    return elements[(n+1)//2-1]